src/autoupdate.c

```diff
@@ -89,47 +109,63 @@ static void policy_warn(void)
 
 enum swupd_code autoupdate_main(int argc, char **argv)
 {
+	char *root_path = NULL;
+
 	if (!parse_options(argc, argv)) {
+		print_help();
 		return SWUPD_INVALID_OPTION;
 	}
-	if (enable && disable) {
-		error("Can not enable and disable at the same time\n");
-		return SWUPD_INVALID_OPTION;
+
+	if (globals.path_prefix) {
+		string_or_die(&root_path, "%s", globals.path_prefix);
```

Comment:
You don't need to duplicate memory for path_prefix. Just use the globals.path_prefix where you used root_path.

Reply:
Okay

Reply:
Done